packages/api/src/domain/medical/cq-patient-data.ts

```diff
@@ -0,0 +1,27 @@
+import { BaseDomain, BaseDomainCreate } from "../base-domain";
+
+export type CQExternalPatient = {
+  patientId: string;
+  systemId: string;
+};
+
+export type CQLinkedGateway = {
+  oid: string;
+  url: string;
+};
+
+export type CQLink = CQExternalPatient & CQLinkedGateway;
+
+// leaving room for other info if needed
+export type CQData = {
+  links: CQLink[];
+};
+
+export type CQDataWithCxId = {
+  cxId: string;
+  data: CQData;
+};
+
+export interface PatientCQDataCreate extends BaseDomainCreate, CQDataWithCxId {}
+
+export interface PatientCQData extends BaseDomain, PatientCQDataCreate {}
```

Comment:
I think this is subjective and depends on how you associate this data. In my mind, it makes more sense to keep `patient` first, since this is very tightly coupled with the patient - this is why it was initially supposed to be in the patient table. And in the future, for XCA requests, we will be querying this table based on the patient ID because this data is patient-centric. 

What do you think?

Reply:
_(sorry for the long answer, but I think its important to share my perspective from diff angles)_

> ...it makes more sense to keep patient first, since this is very tightly coupled with the patient - this is why it was initially supposed to be in the patient table

I see what you mean! And I would 💯 agree with you it this was part of the Metriport logic... but this is specific to CQ. _(it might look like now we're tied to CQ, but that was the case 1 year ago w/ CW, and more HIEs will come)_

It was only originally suggested to store on the patient table b/c that's how CW came to be. And CW was stored on the patient table only b/c of development speed; since the beginning I've been promoting more separation between HIE-specific code/logic from the main logic of the API.

See how easy it is to attach a new HIE to the patient's creation flow, just add another of this specific to CQ on patient create:

```ts
cwCommands.patient.create(...)
```

...and, ideally, this wouldn't even be there it we had event based architecture from the start ([early steps ongoing now](https://metriport.slack.com/archives/C04DMKE9DME/p1702675771527909)).

Had we just coded to get it done, we wouldn't have saved much time and the maintenance of the api from day 0 would have been much more expensive.

Another perspective: imagine in a few months time we (hopefully) have a lot of load on our API and we decide to split services:
- the Metriport API
- a CW integration specific service/app
- a CQ integration specific service/app

It would be easier to move stuff around if we have CQ-related code together and likewise, CW. See OSS' `src/external/commonwell` and `src/external/carequality`, just move the folder around.

And there's the confusion with stuff like `getPatientCQData` - this is super close to other things we do with based off the Patient entity - whereas `getCQPatientData` is clearer as to what it's related to.

The main point is just separation of concerns, and keeping our logic/domain separated to HIE specific.

Reply:
Yea this makes a lot of sense from the separation of concerns pov. Thanks for the detailed write up :)